### ISSUE TITLE
Feature request: Add configurable delay before displaying progress bar

### ISSUE THREAD

**Issue by mattbrooks2010** [MurhafSousli/ngx-progressbar#141]
### Bug Report or Feature Request (mark with an `x`)
```
- [ ] bug report -> please search issues before submitting
- [x] feature request
- [ ] question
```

### Desired functionality
Would it be possible to add support for a configurable delay, in milliseconds, before displaying the progress bar?

The use case I'm targeting is the *automatic plug and play* scenario where your progress bar can be easily implemented within an app for HTTP XHR requests and router events. In many apps, if those types of events complete in less than a few hundred milliseconds, there is no need to display the progress bar to users who generally will not perceive the latency. However, it is very beneficial to show the progress bar for events that take around half a second or longer.

### Other details that might be useful
I have a working implementation, which involves a custom injectable `ProgressService` inspired by the `NgProgress` service class to handle the 'in progress count' and also the delay. A custom `HttpProgressInterceptor` and `RouterProgressModule` are then needed, which make use of the `ProgressService`.

Many thanks for building and sharing this useful component. It would be great if this use case could be handled out of the box.

**Comment by MurhafSousli** [comment]
@mattbrooks2010 Hi can I have a look at the implementation?

**Comment by jasekiw** [comment]
I think this would be simpler with rxjs. https://www.learnrxjs.io/operators/filtering/debounce.html

**Comment by MurhafSousli** [comment]
This feature is available in v4.3.0

**Comment by mattbrooks2010** [comment]
Working great — thanks.
